crates/trie/sparse/src/trie.rs

```diff
@@ -553,6 +572,17 @@ impl RevealedSparseTrie {
         Ok(nodes)
     }
 
+    /// Wipe the trie, removing all values and nodes, and replacing the root with an empty node.
+    pub fn wipe(&mut self) -> SparseTrieResult<()> {
+        let paths = self.values.drain().map(|(k, _)| k).collect::<Vec<_>>();
+        // TODO: can we optimize the batch deletion here?
+        for path in paths {
+            self.remove_leaf(&path)?;
+        }
+        debug_assert_eq!(self.nodes, HashMap::from_iter([(Nibbles::default(), SparseNode::Empty)]));
+        Ok(())
+    }
```

Comment:
mb?
```suggestion
    /// Wipe the trie, removing all values and nodes, and replacing the root with an empty node.
    pub fn wipe(&mut self) {
        *self = Self::default();
    }
```

Reply:
yeah i did it that way at first, but the branch node retaining logic in https://github.com/paradigmxyz/reth/pull/12291 relies on calling the `remove_leaf`. That's why I added this TODO https://github.com/paradigmxyz/reth/blob/e5e052d75dd34ef8dd26999adabbc32f0439aafd/crates/trie/sparse/src/trie.rs#L578

Reply:
@shekhirin but we should just wipe any intermediate node in the database, it's fine if this diverges from the HB behavior

Reply:
we still need to retain the removed branch nodes, I adjusted your code a bit, PTAL

Reply:
nvm, there's a separate flag in updates struct that we can use, so your approach works fine https://github.com/paradigmxyz/reth/blob/7f5fd80cb7d69d1e5d14abf6851111adae762970/crates/trie/db/src/trie_cursor.rs#L126-L129